src/providers/BasicPoolsProvider.tsx

```diff
@@ -176,75 +186,43 @@ export async function getPoolsDataFromRegistry(
-      typeof poolRegistryMultiCall.getA
-    >[] = []
-    const tokenBalancesMultiCalls: ReturnType<
-      typeof poolRegistryMultiCall.getTokenBalances
-    >[] = []
-    const underlyingTokenBalancesMultiCalls: ReturnType<
-      typeof poolRegistryMultiCall.getUnderlyingTokenBalances
-    >[] = []
-    const lpTokenTotalSuppliesMultiCalls: MulticallCall<unknown, BigNumber>[] =
-      []
-
-    registryPoolData.forEach((poolData) => {
-      if (poolData != null) {
-        const { lpToken, poolAddress } = poolData
-        const lpTokenContract = createMultiCallContract<Erc20>(
-          lpToken.toLowerCase(),
-          ERC20_ABI,
-        )
-
-        lpTokenTotalSuppliesMultiCalls.push(lpTokenContract.totalSupply())
-        arePoolsPausedMulticalls.push(
-          poolRegistryMultiCall.getPaused(poolAddress),
-        )
-        virtualPricesMulticalls.push(
-          poolRegistryMultiCall.getVirtualPrice(poolAddress),
-        )
-        swapStoragesMulticalls.push(
-          poolRegistryMultiCall.getSwapStorage(poolAddress),
-        )
-        aParametersMultiCalls.push(poolRegistryMultiCall.getA(poolAddress))
-        tokenBalancesMultiCalls.push(
-          poolRegistryMultiCall.getTokenBalances(poolAddress),
-        )
-        underlyingTokenBalancesMultiCalls.push(
-          poolRegistryMultiCall.getUnderlyingTokenBalances(poolAddress),
-        )
-      }
+    const poolMulticalls = (
+      registryPoolData.filter(Boolean) as {
+        lpToken: string
+        poolAddress: string
+      }[]
+    ).map(({ lpToken, poolAddress }) => {
+      const lpTokenContract = createMultiCallContract<Erc20>(
+        lpToken.toLowerCase(),
+        ERC20_ABI,
+      )
+      return [
+        lpTokenContract.totalSupply(),
+        poolRegistryMultiCall.getA(poolAddress),
+        poolRegistryMultiCall.getPaused(poolAddress),
+        poolRegistryMultiCall.getSwapStorage(poolAddress),
+        poolRegistryMultiCall.getTokenBalances(poolAddress),
+        poolRegistryMultiCall.getUnderlyingTokenBalances(poolAddress),
+        poolRegistryMultiCall.getVirtualPrice(poolAddress),
+      ] as const
     })
 
     const [
+      lpTokensSupplies,
+      aParameters,
       arePoolsPaused,
-      virtualPrices,
       swapStorages,
-      aParameters,
       tokensBalances,
       underlyingTokensBalances,
-      lpTokensSupplies,
+      virtualPrices,
     ] = await Promise.all([
-      ethCallProvider.tryAll(arePoolsPausedMulticalls),
-      ethCallProvider.tryAll(virtualPricesMulticalls),
-      ethCallProvider.tryAll(swapStoragesMulticalls),
-      ethCallProvider.tryAll(aParametersMultiCalls),
-      ethCallProvider.tryAll(tokenBalancesMultiCalls),
-      chunkedTryAll<BigNumber[]>(
-        underlyingTokenBalancesMultiCalls,
-        ethCallProvider,
-        2,
-      ),
-      ethCallProvider.tryAll(lpTokenTotalSuppliesMultiCalls),
+      chunkedTryAll20(poolMulticalls.map((a) => a[0])), // lpTokenSupply
```

Comment:
The one caveat to this approach is that it's important to get the indexes correct.